File: baekjoon/쉬운계단수.cpp
#include <iostream>
using namespace std;

#define MODE 1000000000;
long long dp[101][10];

int main() {

	ios::sync_with_stdio(false);
	cin.tie(NULL);
	int n;
	cin >> n;
	int k = 0;

	for (int i = 1; i < 10; ++i) {
		dp[1][i] = 1;
	}
	for (int i = 2; i <= n; ++i) {
		for (k = 0; k < 10; ++k) {
			if (k == 0) {
				dp[i][k] = dp[i - 1][k + 1];
			}
			else if (k == 9) {
				dp[i][k] = dp[i - 1][k - 1];
			}
			else
				dp[i][k] = (dp[i - 1][k - 1] + dp[i - 1][k + 1])%MODE;
		}
	}
	long long answer = 0;
	for (int i = 0; i < 10; ++i) {
		answer = (answer + dp[n][i]) % MODE;
	}
	cout << answer;
}

File: baekjoon/쉬운계단수.py
n = int(input())
dp = [[0] * 11 for _ in range(101)]
# dp i,j => i개 수, 시작이 j일 때

if n == 1:
    print(9)
else:
    for i in range(10):
        dp[1][i] = 1
    for i in range(2, n + 1):
        dp[i][0] = dp[i - 1][1]
        for j in range(1, 10):
            if j == 9:
                dp[i][j] = dp[i - 1][j - 1]
            else:
                dp[i][j] = (dp[i - 1][j - 1] + dp[i - 1][j + 1])%1000000000
    answer= 0
    for i in range(1, 10):
        answer = (answer + dp[n][i])%1000000000
    print(answer)
